### PR TITLE
Fix host serialization when host.groups is undefined; add tests

### PR DESCRIPTION
**Pull request by kruai**
# Overview

This PR is being created to address the issue we saw in Prod yesterday. If the service attempts to serialize a host for MQ when host.groups does not exist, it raises [an exception](https://kibana.apps.crcp01ue1.o9m8.p1.openshiftapps.com/app/kibana#/discover?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2023-06-08T17:01:01.175Z',to:'2023-06-08T18:50:36.349Z'))&_a=(columns:!(exc_text),filters:!(),index:'43c5fed0-d5ce-11ea-b58c-a7c95afd7a5d',interval:auto,query:(language:lucene,query:'@log_stream:%20%22host-inventory-mq%22%20AND%20levelname:%20ERROR%20AND%20msg:%20%22Unable%20to%20process%20message%22'),sort:!(!('@timestamp',desc)))); this PR fixes that, and adds some thorough testing to make sure it doesn't happen again.

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: tests/test_unit.py [hunk 1]
```diff
@@ -1535,60 +1535,68 @@ def test_with_all_fields(self):
...
+                    "mac_addresses": None,
+                    "ansible_host": None,
+                    "provider_id": None,
+                    "provider_type": None,
+                    **unchanged_data,
+                    "facts": [],
+                    "groups": [],
+                    "tags": [],
+                    "id": str(host_attr_data["id"]),
+                    "created": self._timestamp_to_str(host_attr_data["created_on"]),
```
Comment (thearifismail) [inline tests/test_unit.py:1587]: This should be "created_on" and "modified_on"  because these host properties.  "created" and "updated" are in groups.

**Review by thearifismail — COMMENTED** [review]
Please seen the comment.

File: tests/test_unit.py (hunk 1 above)
Comment (kruai) [reply tests/test_unit.py:1587]: No, `created` and `updated` are correct. I did not change the names of those fields.

**Review by thearifismail — APPROVED** [review]

**Review by fstavela — APPROVED** [review]
LGTM

**Pull request merged**